ui/core/src/model/http.ts

```diff
@@ -11,22 +11,26 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { DataSourceResource } from '@perses-ui/core';
+export interface BasicAuth {
```

Comment:
Do you feel like we can hold off on auth options until we need them? Feels a little premature to me. (I suspect we'll probably never want `BasicAuth` unless someone requests it, but 🤷‍♂️ ).

Reply:
The auth options are just here for the future form of the datasource in the UI. I can remove it for sure.
It's only used in the backend, but by looking about how people are protecting Prometheus, `BasicAuth` is maybe the simplest way. So pretty sure it will be required directly.
And it's already considered / handled in the Perses backend here: https://github.com/perses/perses/blob/main/internal/api/core/middleware/proxy.go#L205-L218